_posts/2024-03-19-credential-stores-on-client-side.adoc

```diff
@@ -0,0 +1,155 @@
+```
+Now let's store the login information for our `quickStartUser` identity into the credential-store: 
+```
+ $ WILDFLY_HOME/bin/elytron-tool.sh credential-store --location "/PATH/TO/mycredstore.cs" --password StorePassword --add quickstartUser --secret quickstartPwd1!
+```
+Our WildFly client will be able to access this credential-store to access the login information for the identity. Here the quickstartUser is an alias that the password is being stored under and it does not have to match the identity name we specified earlier.
+
+== Configure wildfly-config.xml
+Now that we have our credential-store set up, we need to let our WildFly Client know where to find it and how to access it. We will be making use of a file called `wildfly-config.xml` to communicate that. This file can be found inside the `src/main/resources` folder. 
+
+We can specify the location and the type and password for the credential-store as follows: 
+```
+<credential-stores>
+    <credential-store name="mycredstore">
+        <attributes>
+            <attribute name="keyStoreType" value="JCEKS"/>
+            <attribute name="location" value="PATH/TO/mycredstore.cs"></attribute>
+        </attributes>
+            <protection-parameter-credentials>
+                <clear-password password="StorePassword"/>
+        </protection-parameter-credentials>
+    </credential-store>
+</credential-stores>
+```
+This should be added into the xml file already, but you may need to update the paths with the actual paths for the credential store. Note you would need to replace "PATH/TO/mycredstore.cs" with the actual path to your credential store.
+
+Now we need to specify which identity to extract from our credential-store. A `credential-store` can hold multiple credentials that are used for various purposes. We can specify which one to use by identifying it with the unique alias we declared earlier. We will also be specifying the authentication mechanism here to match what we specified for the server: 
+```
+<authentication-configurations>
+    <configuration name="default-config">
+        <set-user-name name="quickstartUser"/>
+        <credentials>
+            <credential-store-reference store="mycredstore" alias="quickstartUser"/>
+        </credentials>
+        <sasl-mechanism-selector selector="SCRAM-SHA-512"/>
+        <providers>
+            <use-service-loader />
+        </providers>
+    </configuration>
+</authentication-configurations>
+```
+== Build and Deploy the application
+Now we can use maven to build and deploy the application using the commands below: 
+```
+    $ mvn clean install wildfly:deploy
+``` 
+
+Unlike other applications, this will not be a web application, rather something we can run on the terminal. If you examine the server logs, you will notice that instead of a `.war` deployment, we will be producing a `.jar` deployment file. We can access the application by running the client using the command below: 
```

Comment:
hm just a total minor suggestion that you can ignore, but since we are not mentioning other applications in this blog, and there are many kinds of applications other than web applications, and we can access `.jar` with other apps or code and not just with the terminal, I would just write:

This will not be a web application, so we won't use a browser to access it, but instead we will use the terminal.